src/v/cluster/non_replicable_partition_manager.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "cluster/non_replicable_partition.h"
+#include "cluster/partition.h"
+#include "storage/fwd.h"
+
+#include <seastar/core/gate.hh>
+
+#include <absl/container/flat_hash_map.h>
+
+namespace cluster {
+
+class non_replicable_partition_manager {
```

Comment:
is this something that can go in `src/v/coproc`? this seems like it is going add a lot of complexity. why aren't we working to generalize the existing abstractions to handle these new cases?

Reply:
It cannot because `controller_backend` calls `manage` and `remove`

Reply:
Doing it now :D